contract/sep-nft-collection.md

```diff
@@ -0,0 +1,114 @@
+## Summary
+
+This proposal introduces a standard interface for creating and managing NFT collections on Soroban, ensuring compatibility and interoperability within the Stellar ecosystem, and aligning with the previously defined NFT interface and the ERC-1155 standard.
+
+## Motivation
+
+A standardized interface for NFT collections on Soroban will facilitate seamless interaction within the Stellar ecosystem, enhancing user experience and developer interaction by providing a consistent API for NFT collection creation, management, and interaction, while also allowing for efficient batch transfers and management of both fungible and non-fungible tokens within a single contract.
+
+## Abstract
+
+This proposal defines a Rust trait for NFT collections, ensuring that all NFT collection contracts on Soroban adhere to a common API. This API includes functions for creating new NFTs within a collection, transferring NFTs, querying for ownership, and for accessing metadata necessary for displaying NFT information in wallets and other user interfaces.
+
+## Specification
+
+```rust
+pub trait NFTCollectionFactory {
```

Comment:
Why a factory? Doesn't that imply that it's deploying contracts?

Reply:
It is deploying contracts. I was unable to find a way to make them compatible with token interface without having seperate addresses for minted token. 

Appreciate the feedback. Definitely not saying this is the best or correct way but rather a starting point for us to build off of.

Reply:
Is the idea here that every NFT is a separate contract instance? If so, what is the purpose/value of having a contract that allows managing NFT collection as a whole? Intuitively, I'd think that the purpose of collection contract is to store and manage multiple NFTs without deploying separate contracts for each one (this is also reflected in functions like `transfer` that take `token_id` as an input). However if this just duplicates the functionality of regular (singular) tokens, then it seems redundant - the collection management would better be managed off-chain.

To be clear, doing something like `transfer` when token is a separate contract would require auth for `token.transfer`, i.e. `transfer` would act just as a router that maps `token_id` to and `Address`.  Or is the idea to have factory as admin for NFTs, so that `transfer` would do clawback+mint on the underlying token?